extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/image_2_text.py

```diff
@@ -232,15 +235,23 @@ def validate_and_retrieve_image_from_attachments(prompt: Prompt) -> list[str]:
     if not hasattr(prompt.input, "attachments") or len(prompt.input.attachments) == 0:
         raise ValueError(f"No attachments found in input for prompt {prompt.name}. Please add an image attachment to the prompt input.")
 
-    image_uris: list[str] = []
+    images: list[Union[str, Image]] = []
 
     for i, attachment in enumerate(prompt.input.attachments):
         validate_attachment_type_is_image(attachment)
 
-        if not isinstance(attachment.data, str):
+        input_data = attachment.data
+        if not isinstance(input_data, str):
             # See todo above, but for now only support uri's
             raise ValueError(f"Attachment #{i} data is not a uri. Please specify a uri for the image attachment in prompt {prompt.name}.")
 
-        image_uris.append(attachment.data)
+        # Really basic heurestic to check if the data is a base64 encoded str
+        # vs. uri. This will be fixed once we have standardized inputs
+        # See https://github.com/lastmile-ai/aiconfig/issues/829
+        if len(input_data) > 10000:
```

Comment:
This seems reasonable for now until we standardize input types.

Would it be better to flip it? i.e. check if the input_data is a URL or file path?

Perhaps something like:

```
from urllib.parse import urlparse
from pathlib import Path

def could_be_url_or_path(s):
    # First, check if it could be a URL.
    try:
        urlparse(s)
    except ValueError:
        pass  # Not a URL. Might still be a path.
    else:
        # URL parsing didn't raise an error, so s might be a URL.
        # Check that s has a scheme and a network location.
        if all([urlparse(s).scheme, urlparse(s).netloc]) or s.startswith('file://'):
            return True  # URL

    # Next, check if it could be a path
    p = Path(s)
    if p.is_absolute() or (p.is_relative_to('.') and p.suffix):
        return True  # Path

    # It's not any of them above -> Return False
    return False
```